File: Fortran/IBM/execute/F2003/mxminAll/var/maxMin/mxminVarBindC3.f90
!#######################################################################
!*  ===================================================================
!*  XL Fortran Test Case                          IBM INTERNAL USE ONLY
!*  ===================================================================
!*  ===================================================================
!*
!*  TEST CASE TITLE            :
!*
!*  PROGRAMMER                 : William Zhang 
!*  DATE                       : 2/05/2006
!*  ORIGIN                     : AIX Compiler Development, Toronto Lab
!*                             :
!*
!*  PRIMARY FUNCTIONS TESTED   : Section 13.7.71[3,4,6,8,9]:
!*                               character argument for MAX*/MIN* intrinsics 
!*                             :
!*  SECONDARY FUNCTIONS TESTED : 
!*
!*
!*  DESCRIPTION                : MAX/MIN with variable as argument 
!*                               while used as array element in interlanguage 
!*                               calls with C. 
!* ===================================================================

module modbnd2

     interface
        subroutine sub1(arg) bind(c)
            use  ISO_C_BINDING
            character(kind=C_CHAR) :: arg(5)
        end subroutine
     end interface

end module modbnd2

module modbnd22

     interface
        subroutine sub2(arg) bind(c)
            use  ISO_C_BINDING
            character(kind=C_CHAR) :: arg(5)
        end subroutine
     end interface

     procedure(sub2), pointer :: procptr

end module modbnd22

program mxminVarBindC3 

    use  ISO_C_BINDING

    use modbnd2
    use modbnd22

    character*2 tst1, tst2, tst3, tst4, tst5
    tst1 = "dd"
    tst2 = "xx"
    tst3 = "aa"
    tst4 = "ff"
    tst5 = "hh"

    procptr => sub1

    call procptr((/max(tst3,tst4)//C_NULL_CHAR, min(tst1, tst3, "zz")//C_NULL_CHAR, max(tst2, "d")//C_NULL_CHAR, min(tst1, "aa")//C_NULL_CHAR, tst5//C_NULL_CHAR/))

end program mxminVarBindC3 

File: Fortran/IBM/execute/F2003/mxminAll/var/maxMin/mxminVarBindC3.c
void sub1 ( char cc[5][3] )
{
   int i, j;

   if(cc[0][0] != 'f') exit(31);
   if(cc[0][1] != 'f') exit(32);
   if(cc[0][2] != '\0') exit(33);

   if(cc[1][0] != 'a') exit(34);
   if(cc[1][1] != 'a') exit(35);
   if(cc[1][2] != '\0') exit(36);

   if(cc[2][0] != 'x') exit(37);
   if(cc[2][1] != 'x') exit(38);
   if(cc[2][2] != '\0') exit(39);

   if(cc[3][0] != 'a') exit(41);
   if(cc[3][1] != 'a') exit(42);
   if(cc[3][2] != '\0') exit(43);

   if(cc[4][0] != 'h') exit(44);
   if(cc[4][1] != 'h') exit(45);
   if(cc[4][2] != '\0') exit(46);

}

